### PR TITLE
Avoid calling as_tensor twice

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112866

Sometimes doing so may copy and that's not good. We avoid that by
setting global flags.

cc @mruberry @rgommers @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_numpy/_ndarray.py [hunk 1]
```diff
@@ -494,7 +494,7 @@ def array(obj, dtype=None, *, copy=True, order="K", subok=False, ndmin=0, like=N
     if like is not None:
         raise NotImplementedError("'like' parameter is not supported.")
     if order != "K":
-        raise NotImplementedError
+        raise NotImplementedError()
```
Comment (kit1980) [inline torch/_numpy/_ndarray.py:497]: Thanks for catching this.
There must be a linter for such things, maybe we do have it but don't lint _numpy?

Comment (kit1980) [reply torch/_numpy/_ndarray.py:497]: cc @Skylion007 as a linting enthusiast :-D

**Review by kit1980 — APPROVED** [review]

**Review by ev-br — APPROVED** [review]
LGTM

**Pull request closed without merging**